### DISCUSSION TITLE
index-only option for simple search

### DISCUSSION THREAD

**Discussion by fossildoc** [dbeaver/dbeaver#14303] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am a heavy user of the simple search, the one I get when I type ctrl-f. The items I search for are always in an indexed field of my database, but the target text appears many times in other fields. This substantially increases the time it takes for me to find anything.

**Describe the solution you'd like**
The DB Full--Text search window contains a very useful option, Fast search (indexed). I would like that option added to the ctrl-f Find/Replace window. The search would then be conducted only on indexed fields.

**Describe alternatives you've considered**
DB Full-Text is not a viable alternative because the Databases information must be entered each time, which consumes even more time.

**Comment by Matvey16** [comment]
@fossildoc Can you elaborate how you see this added to find/replace? That is a basic cheap feature and it performs the search only in fetched rows.

**Comment by fossildoc** [comment]
Fetched rows has nothing to do with it.

"DB Full-Text" has a checkbox,"Fast search (indexed)". This causes the
search to be made only in indexed fields of the table specified in the
"Databases" dropdown list. If the box is not checked, the search is made in
all the fields of the specified table.

That feature is not available in the ctrl-f "Find/Replace" window, so
searches are always made in all the fields of the displayed table.

I would like a "Fast search (indexed)" option added to the "Find/Replace"
window so that the search is made only in indexed fields of the displayed
table.

**Comment by Matvey16** [comment]
@fossildoc I was talking about fetched rows because the "Find/Replace" search is intended for them. They are different operations. DB Full-text performs a query on your database, so indexed search lowers the cost of that query. But the search with <kbd>ctrl+f</kbd> is performed on the client only on the fields that are fetched currently in data editor and doesn't actually interact with your database, so it doesn't need an indexed search.
Maybe we are talking about different things, but Find/Replace is just a basic client-side search that you can find in any text editor like notepad.

**Comment by fossildoc** [comment]
For the life of me I cannot fathom what you don't understand about my
request. I don't see any ambiguity in it. Let's go through it one more time
with feeling. I'll speak slowly and loudly, like I was talking to a foreign
tourist.

CTRL-F FIND/REPLACE WILL RETURN A SUBSET OF THE FETCHED ROWS OF THE
CURRENTLY DISPLAYED TABLE FOR WHICH THE SEARCH TARGET IS FOUND IN ANY OF
THE DISPLAYED COLUMNS.

I WANT AN OPTION THAT WILL RETURN A SUBSET OF THE FETCHED ROWS OF THE
CURRENTLY DISPLAYED TABLE FOR WHICH THE SEARCH TARGET IS FOUND ONLY IN
DISPLAYED COLUMNS WHICH ARE INDEXED.

If you still cannot understand this, please pass my request to an English
native speaker with experience in database design.

**Comment by Matvey16** [comment]
@fossildoc I don't understand why you think that ctrl+f has anything to do with database design or table metadata (index in this case). As I already said this search has nothing to do with any metadata or indexes, it doesn't even return a subset of rows, it returns a subset of table cells that contain alphanumeric value that you specified in your search. I want to again point on the fact that this search doesn't read any table metadata or even actual data from the table, it works only with the data that is already fetched for the client side. Here is the example of how it works for a table with 1 million rows, but at the time of the actual search it has only 100 rows fetched

![z5wSNUxr4Y](https://user-images.githubusercontent.com/82543000/134161054-be640ddd-dd41-422e-9e10-4a82f379c26d.gif)

For this search to even know which column is indexed it would require an additional metadata read. And that can be a redundant option for such a basic feature and that is why we have DB Full-Text search

The only feature that I see fitting here is adding an option to search only in selected columns. Do you think it will be enough?

**Comment by Matvey16** [comment]
New ticket opened based on this feature request #15070
